### ISSUE TITLE
pact-jvm-server v3.6.3 onwards: Exception when request contains cookie

### ISSUE THREAD

**Issue by hstamminger** [DiUS/pact-jvm#928]
java: 1.8.0, 11, 12
pact-jvm-server: 3.6.3+ 
command-line: `bin\pact-jvm-server_2.12.bat --debug --pact-version 3`

We get an exception in the pact-jvm-server (resulting in 500 response) when a request contains a cookie. The last working version is 3.6.2. 

---

Example:
**Request 'create mock'**
`curl --request POST \
  --url 'http://localhost:29999/create?state=Basistestfixture&path=' \
  --header 'content-type: application/json' \
  --data '{ "consumer": { "name": "pact-consumer" }, "provider": { "name": "pact-provider" }, "interactions": [ { "description": "get time", "request": { "method": "GET", "path": "/resource/time/current", "query": { "locale": [ "de" ] } }, "response": { "body": { "time": "13:00:34.890" }, "status": 200 }, "providerStates": [ { "name": "Basistestfixture" } ] } ], "metadata": { "pact-specification": { "version": "3.0.0" } } }'`

**Response 'create mock':**
{"port": 27537}

**Request to created mock:** 
`curl --request GET 
  --url 'http://localhost:34228/resource/time/current?locale=de' 
  --header 'cookie: test=12345'

**Response:** 500 - Internal Server Error

Exception in pact-jvm-server:
```
Exception caught handling request:
groovy.lang.MissingMethodException: No signature of method: scala.collection.convert.Wrappers$SeqWrapper.split() is applicable for argument types: (String) values: [;]
Possible solutions: split(groovy.lang.Closure), split(groovy.lang.Closure), split(groovy.lang.Closure), wait(), init(), init()
        at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.unwrap(ScriptBytecodeAdapter.java:70)
        at org.codehaus.groovy.vmplugin.v7.IndyGuardsFiltersAndSignatures.unwrap(IndyGuardsFiltersAndSignatures.java:175)
        at org.codehaus.groovy.vmplugin.v7.IndyInterface.selectMethod(IndyInterface.java:234)
        at au.com.dius.pact.model.Request.cookie(Request.groovy:92)
        at au.com.dius.pact.model.RequestMatching$.requestMismatches(RequestMatching.scala:55)
        at au.com.dius.pact.model.RequestMatching$.compareRequest(RequestMatching.scala:44)
        at au.com.dius.pact.model.RequestMatching.compareToActual$1(RequestMatching.scala:12)
        at au.com.dius.pact.model.RequestMatching.$anonfun$matchInteraction$1(RequestMatching.scala:13)
        at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:233)
        at scala.collection.Iterator.foreach(Iterator.scala:937)
        at scala.collection.Iterator.foreach$(Iterator.scala:937)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1425)
        at scala.collection.IterableLike.foreach(IterableLike.scala:70)
        at scala.collection.IterableLike.foreach$(IterableLike.scala:69)
        at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
        at scala.collection.TraversableLike.map(TraversableLike.scala:233)
        at scala.collection.TraversableLike.map$(TraversableLike.scala:226)
        at scala.collection.AbstractTraversable.map(Traversable.scala:104)
        at au.com.dius.pact.model.RequestMatching.matchInteraction(RequestMatching.scala:13)
        at au.com.dius.pact.consumer.PactSession.receiveRequest(PactSession.scala:50)
        at au.com.dius.pact.consumer.StatefulMockProvider.handleRequest(MockProvider.scala:74)
        at au.com.dius.pact.consumer.UnfilteredMockProvider$Routes$$anonfun$intent$1.applyOrElse(UnfilteredMockProvider.scala:24)
        at au.com.dius.pact.consumer.UnfilteredMockProvider$Routes$$anonfun$intent$1.applyOrElse(UnfilteredMockProvider.scala:23)
        at scala.PartialFunction$Lifted.apply(PartialFunction.scala:224)
        at scala.PartialFunction$Lifted.apply(PartialFunction.scala:220)
        at unfiltered.response.Pass$PartialAttempt.attemptWithPass(pass.scala:72)
        at unfiltered.response.Pass$PassingAttempt.attempt(pass.scala:65)
        at unfiltered.response.Pass$PassingAttempt.attempt$(pass.scala:65)
        at unfiltered.response.Pass$PartialAttempt.attempt(pass.scala:67)
        at unfiltered.response.Pass$.$anonfun$fold$1(pass.scala:48)
        at unfiltered.response.Pass$FunctionAttempt.apply(pass.scala:77)
        at unfiltered.netty.cycle.Plan.$anonfun$channelRead$3(plans.scala:68)
        at unfiltered.netty.cycle.Plan.catching(plans.scala:41)
        at unfiltered.netty.cycle.Plan.catching$(plans.scala:40)
        at au.com.dius.pact.consumer.UnfilteredMockProvider$Routes$.catching(UnfilteredMockProvider.scala:19)
        at unfiltered.netty.cycle.Plan.$anonfun$channelRead$2(plans.scala:67)
        at unfiltered.netty.cycle.SynchronousExecution.executeIntent(deferrals.scala:23)
        at unfiltered.netty.cycle.SynchronousExecution.executeIntent$(deferrals.scala:23)
        at au.com.dius.pact.consumer.UnfilteredMockProvider$Routes$.executeIntent(UnfilteredMockProvider.scala:19)
        at unfiltered.netty.cycle.Plan.$anonfun$channelRead$1(plans.scala:66)
        at unfiltered.netty.cycle.Plan.catching(plans.scala:41)
        at unfiltered.netty.cycle.Plan.catching$(plans.scala:40)
        at au.com.dius.pact.consumer.UnfilteredMockProvider$Routes$.catching(UnfilteredMockProvider.scala:19)
        at unfiltered.netty.cycle.Plan.channelRead(plans.scala:65)
        at unfiltered.netty.cycle.Plan.channelRead$(plans.scala:61)
        at au.com.dius.pact.consumer.UnfilteredMockProvider$Routes$.channelRead(UnfilteredMockProvider.scala:19)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:102)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:323)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:297)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1434)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:965)
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:644)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:579)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:496)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:458)
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:897)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.lang.Thread.run(Thread.java:748)
```

---
Complete logfile from pact-jvm-server: [pact-jvm-server.log](https://github.com/DiUS/pact-jvm/files/3554700/pact-jvm-server.log)

**Comment by uglyog** [comment]
Fix has been released with version 3.6.13

**Comment by hstamminger** [comment]
Hi thanks. It's working now.
